### PR TITLE
feat: React router migration

### PR DESCRIPTION
**Pull request by mmelko**
This is a WIP PR. I've currently migrated and refactored all (I hope) routes in all plugins except camel.

Thinks to do: 
- migrate camel plugin,
- configure local dev environment to auto redirect to [host]/hawtio, this was removed in router v6
@tadayosi  please check if applicable so far

### CODE REVIEW

File: packages/hawtio/src/plugins/jmx/JmxContent.tsx [hunk 1]
```diff
@@ -63,20 +78,22 @@ export const JmxContent: React.FunctionComponent = () => {
       <PageSection>
         {node.objectName &&
           <React.Fragment>
-            {mbeanRoutes}
-            <Route key='root' exact path='/jmx' component={Attributes} />
+            <Routes>
+              {mbeanRoutes}
+              <Route key="root" path="/*" element={<Navigate to={"attributes"}/>}/>
```
Comment (tadayosi) [inline packages/hawtio/src/plugins/jmx/JmxContent.tsx:83]: with this path setting, when accessing http://localhost:3000/hawtio/?con=springboot from Connect it redirects to http://localhost:3000/hawtio/jmx/*?con=springboot (note `*`). Why can't it be `/`?

File: packages/hawtio/src/plugins/jmx/index.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ export const jmx = () => {
   hawtio.addPlugin({
     id: 'jmx',
     title: 'JMX',
-    path: '/jmx',
+    path: 'jmx/*',
```
Comment (tadayosi) [inline packages/hawtio/src/plugins/jmx/index.ts:11]: if react router v6 requires `/*` it shouldn't be configured here in each plugin. it should be a ui page concern so should be implemented in `HawtioPage.tsx`. A plugin shouldn't be aware of such impl details.

File: packages/hawtio/src/preferences/HawtioPreferences.tsx [hunk 3]
```diff
@@ -35,15 +34,15 @@ export const HawtioPreferences: React.FunctionComponent = () => {
...
               <Route
                 key={prefs.id}
-                path={path(prefs.id)}
-                component={prefs.component} />
+                path={prefs.id}
+                element={React.createElement(prefs.component)} />
             ))}
-            <Redirect exact from='/preferences' to={path('home')} />
-          </Switch>
+          <Route path='/' element ={<Navigate  to={'home'} />} />
```
Comment (tadayosi) [inline packages/hawtio/src/preferences/HawtioPreferences.tsx:44]: `element =` -> `element=`

File: packages/hawtio/src/ui/page/HawtioPage.tsx [hunk 4]
```diff
@@ -48,25 +48,20 @@ export const HawtioPage: React.FunctionComponent = () => {
...
+        <Routes>
           {/* plugins */}
           {plugins.map(plugin => (
             <Route
               key={plugin.id}
               path={plugin.path}
-              component={plugin.component}
-            />
+              element= {React.createElement(plugin.component)}>
+              </Route>
```
Comment (tadayosi) [inline packages/hawtio/src/ui/page/HawtioPage.tsx:58]: why not use `/>`?

**Review by tadayosi — CHANGES_REQUESTED** [review]
there are minor comments but overall it looks great. thanks!

**Comment by tadayosi** [comment]
> * configure local dev environment to auto redirect to [host]/hawtio, this was removed in router v6

Playing with this pull req, I don't see this issue is so severe to fix before merging the fix. It's just a dev env issue and developers can bookmark http://localhost:3000/hawtio/ instead of http://localhost:3000/ to workaround it.

What seems more important is a code formatting issue, since there have been a lot of changes regarding code formatting in this pull req. Which IDE / code formatter do you use?  VS Code and Prettier?  I feel that we've come to the point where we need to use the same code formatter, because if each person uses their own formatter I wouldn't like to see a lot of changes on just formatting every time in the future.

File: packages/hawtio/src/help/HawtioHelp.tsx [hunk 5]
```diff
@@ -30,19 +42,19 @@ export const HawtioHelp: React.FunctionComponent = () => {
...
         <Card isFullHeight>
-          <Switch>
+          <Routes>
             {helpRegistry.getHelps().map(help =>
-              <Route key={help.id} path={path(help.id)}>
+              <Route key={help.id} path={path(help.id)} element={
                 <CardBody>
                   <TextContent>
                     <Markdown>{help.content}</Markdown>
+                    blabla
```
[the remark this answers is not in the document]
Comment (mmelko) [reply packages/hawtio/src/help/HawtioHelp.tsx:51]: ooops ..

**Comment by mmelko** [comment]
@tadayosi I fixed styling issues (I hope) and incorporated your comments, please check.

File: packages/hawtio/src/help/HawtioHelp.tsx [hunk 6]
```diff
@@ -1,15 +1,14 @@
 import { Card, CardBody, Nav, NavItem, NavList, PageGroup, PageNavigation, PageSection, PageSectionVariants, TextContent, Title } from '@patternfly/react-core'
 import React from 'react'
 import Markdown from 'react-markdown'
-import { NavLink, Redirect, Route, Switch, useLocation } from 'react-router-dom'
+import { NavLink, Navigate, Route, Routes, useLocation } from 'react-router-dom'
 import help from './help.md'
-import { helpRegistry } from './registry'
+import {helpRegistry} from './registry'
```
Comment (tadayosi) [inline packages/hawtio/src/help/HawtioHelp.tsx:6]: unnecessary code change

File: packages/hawtio/src/plugins/jmx/JmxContent.tsx [hunk 7]
```diff
@@ -63,20 +61,22 @@ export const JmxContent: React.FunctionComponent = () => {
...
-            <Route key='root' exact path='/jmx' component={Attributes} />
+            <Routes>
+              {mbeanRoutes}
+              <Route key="root" path="/" element={<Navigate to={"attributes"}/>}/>
+            </Routes>
           </React.Fragment>
         }
         {!node.objectName &&
-          <JmxContentMBeans />
+          <JmxContentMBeans/>
```
Comment (tadayosi) [inline packages/hawtio/src/plugins/jmx/JmxContent.tsx:71]: unnecessary code change

File: packages/hawtio/src/plugins/jmx/JmxContent.tsx [hunk 8]
```diff
@@ -63,20 +61,22 @@ export const JmxContent: React.FunctionComponent = () => {
...
         }
       </PageSection>
     </React.Fragment>
   )
 }
 
 const JmxContentMBeans: React.FunctionComponent = () => {
-  const { node, setNode } = useContext(MBeanTreeContext)
+  const {node, setNode} = useContext(MBeanTreeContext)
```
Comment (tadayosi) [inline packages/hawtio/src/plugins/jmx/JmxContent.tsx:79]: unnecessary code change

File: packages/hawtio/src/plugins/jmx/JmxContent.tsx [hunk 9]
```diff
@@ -91,7 +91,7 @@ const JmxContentMBeans: React.FunctionComponent = () => {
       <Card>
         <CardBody>
           <Text component="p">
-            <InfoCircleIcon /> This node has no MBeans.
+            <InfoCircleIcon/> This node has no MBeans.
```
Comment (tadayosi) [inline packages/hawtio/src/plugins/jmx/JmxContent.tsx:94]: unnecessary code change

File: packages/hawtio/src/plugins/jmx/JmxContent.tsx [hunk 10]
```diff
@@ -114,8 +114,8 @@ const JmxContentMBeans: React.FunctionComponent = () => {
         cells={columns}
         rows={rows}
       >
-        <TableHeader />
-        <TableBody onRowClick={selectChild} />
+        <TableHeader/>
+        <TableBody onRowClick={selectChild}/>
```
Comment (tadayosi) [inline packages/hawtio/src/plugins/jmx/JmxContent.tsx:118]: unnecessary code change

**Review by tadayosi — CHANGES_REQUESTED** [review]
overall it looks great but there are still a few unnecessary code changes.

**Review by tadayosi — APPROVED** [review]
LGTM now, thanks!

**Comment by tadayosi** [comment]
@mmelko Once conflicts resolved, can we merge this one?

**Pull request merged**